examples/test_client2/test.py

```diff
@@ -78,9 +78,9 @@ def run_tests():
     
     # Set consumer key, consumer secret, and server base URL from user input or
     # use default values.
-    CONSUMER_KEY = raw_input("consumer key: ") or CONSUMER_KEY
-    CONSUMER_SECRET = raw_input("consumer secret: ") or CONSUMER_SECRET
-    SERVER_URL = raw_input("server base url: ") or SERVER_URL
+    CONSUMER_KEY = CONSUMER_KEY or raw_input("consumer key: ")
```

Comment:
I don't understand these changes.  The whole point of the way it was before is you can override the default, or hit enter to accept the default.  Your change makes it so it's not possible to override the default (without changing the source).  Is that what you intended?

I'm guessing this is another strings-vs-bytes thing, in which case the right thing to do is to cast one or the other of these appropriately.

Reply:
My intention was that since the `CONSUMER_KEY` and `CONSUMER_SECRET` default to `''`, that the user could either edit the source code to add the keys (and my changes would use the keys that the user added without asking for input) or the user could leave it as is (and the user would be asked for input).

Reply:
Interesting.  I didn't write this original code, but I think the intent was the other way around: you could edit the source to set those to some installation-specific default, but even if you did that you'd still have the opportunity to override it at the prompt.  Right now, hitting <enter> at the prompt won't do anything useful, so the code works the same either way, but I think the intended semantics are the way the code was before your change.